pkgs/by-name/fe/fedimint/package.nix

```
@@ -0,0 +1,98 @@
+{ lib
+, fetchFromGitHub
+, openssl
+, pkg-config
+, protobuf
+, rustPlatform
+, stdenv
+, libiconv
+, clang
+, libclang
+, Security
+, SystemConfiguration
+}:
+let
+  gitRev = "bcc85e8201615b464f9369371d6640823ff12a1a";
+in
+rustPlatform.buildRustPackage {
+  pname = "fedimint";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "fedimint";
+    repo = "fedimint";
+    rev = gitRev;
+    hash = "sha256-eOGmx/freDQLxZ48nP9Y2kA84F6sdig6qfZwnJfOB3g=";
+  };
+
+  cargoHash = "sha256-qK9IvsVLoppwa+BkOi2RqrBY3V97tXS8tZz5Rq7ilH8=";
+
+  nativeBuildInputs = [
+    protobuf
+    pkg-config
+    clang
+    libclang.lib
+  ];
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [
+    Security
+    libiconv
+    Security
+    SystemConfiguration
+  ];
+
+  outputs = [ "out" "fedimintCli" "fedimint" "gateway" "gatewayCli" "devimint" ];
+
+  postInstall = ''
+    mkdir -p $fedimint/bin
+    mkdir -p $fedimintCli/bin
+    mkdir -p $gateway/bin
+    mkdir -p $gatewayCli/bin
+    mkdir -p $devimint/bin
+
+    # delete fuzzing targets and other binaries no one cares about
+    binsToKeep=(fedimint-cli fedimint-dbtool recoverytool fedimintd gatewayd gateway-cli gateway-cln-extension devimint)
+    keepPattern=$(printf "|%s" "''${binsToKeep[@]}")
+    keepPattern=''${keepPattern:1}
+    find "$out/bin" -maxdepth 1 -type f | grep -Ev "(''${keepPattern})" | xargs rm -f
+
+    # fix the upstream name
+    mv $out/bin/recoverytool $out/bin/fedimint-recoverytool
+
+
+    cp -a $releaseDir/fedimint-cli  $fedimintCli/bin/
+    cp -a $releaseDir/fedimint-dbtool  $fedimintCli/bin/
+    cp -a $releaseDir/recoverytool  $fedimintCli/bin/fedimint-recoverytool
+
+    cp -a $releaseDir/fedimintd  $fedimint/bin/
+
+    cp -a $releaseDir/gateway-cli $gatewayCli/bin/
+
+    cp -a $releaseDir/gatewayd $gateway/bin/
+    cp -a $releaseDir/gateway-cln-extension $gateway/bin/
+
+    cp -a $releaseDir/devimint $devimint/bin/
+  '';
+
+  FEDIMINT_BUILD_FORCE_GIT_HASH = gitRev;
+
+  PROTOC = "${protobuf}/bin/protoc";
+  PROTOC_INCLUDE = "${protobuf}/include";
+  OPENSSL_DIR = "${openssl.dev}";
+  LIBCLANG_PATH = "${libclang.lib}/lib";
+
+
+  # currently broken, will require some upstream fixes
```

```suggestion

  # currently broken, will require some upstream fixes
```
Everything or only partly?

---

Small fix for a lack of HOME, but I need to find the time to verify. I'm also not sure how valuable is running our unit-tests in nixpkgs at all. We have a whole test suite with integration and e2e tests, and our unit-tests are comparatively low value in a grand scheme of things.